tests/components/zwave_js/test_init.py

```diff
@@ -404,6 +410,94 @@ async def test_unique_id_migration_notification_binary_sensor(
     entity_entry = ent_reg.async_get(NOTIFICATION_MOTION_BINARY_SENSOR)
     new_unique_id = f"{client.driver.controller.home_id}.52-113-0-Home Security-Motion sensor status.8"
     assert entity_entry.unique_id == new_unique_id
+    assert ent_reg.async_get_entity_id("binary_sensor", DOMAIN, old_unique_id) is None
+
+
+async def test_old_entity_migration(
+    hass, hank_binary_switch_state, client, integration
+):
+    """Test old entity on a different endpoint is migrated to a new one."""
+    node = Node(client, hank_binary_switch_state)
+
+    ent_reg = er.async_get(hass)
+    dev_reg = dr.async_get(hass)
+    device = dev_reg.async_get_or_create(
+        config_entry_id=integration.entry_id, identifiers={get_device_id(client, node)}
+    )
+
+    SENSOR_NAME = "sensor.smart_plug_with_two_usb_ports_value_electric_consumed"
+    entity_name = SENSOR_NAME.split(".")[1]
+
+    # Create entity RegistryEntry using old unique ID format
+    old_unique_id = f"{client.driver.controller.home_id}.32-50-1-value-66049"
+    entity_entry = ent_reg.async_get_or_create(
+        "sensor",
+        DOMAIN,
+        old_unique_id,
+        suggested_object_id=entity_name,
+        config_entry=integration,
+        original_name=entity_name,
+        device_id=device.id,
+    )
+    assert entity_entry.entity_id == SENSOR_NAME
+    assert entity_entry.unique_id == old_unique_id
+
+    # Do this twice to make sure re-interview doesn't do anything weird
```

Comment:
We're not hitting the continue with re-interview comment in the code.

Reply:
Are you thinking we don't need it or do you think we need a test to hit that logic? I actually was doing some testing and I don't think that we send another node ready event on re-interview, so we're currently not handling changes that occur during a reinterview. Something to look into later

Reply:
If it's a case that needs to be handled we should try to test it.

Reply:
Turns out there was a bug in my logic because of my use of `defaultdict` so I'm glad you mentioned this!